### ISSUE TITLE
Feature: Allow collapsing cells in the signature

### ISSUE THREAD

**Issue by jamievicary** [jamievicary/globular#8]
Add a 'minimize' icon to a cell's entry in the signature. Clicking this once will make the cell's entry collapse to a thin horizontal line, taking up little vertical space, showing only the cell's name and the toggle button to re-expand it.

Minimized cells should not be candidates for matching when clicking on a diagram. They are auxiliary elements of the signature, which serve a structural purpose for building up the signature, but no longer serve an active purpose.

**Comment by cdgls** [comment]
I think perhaps minimized cells should still be candidates for matching when clicking on the diagram.  Sometimes you want to minimize auxiliary cells, as you say, but sometimes you just want to minimize cells to save space or focus your attention on a particular collection of cells, but you still need to use the other minimized cells in constructions.

**Comment by jamievicary** [comment]
So there are two things we might want to do: *disable* a cell, meaning "I still need this formally in my signature, but I don't want to consider it for future matches", and *hide* a cell, meaning "I'm bored of this taking up space in my signature, so make it really tiny, but still allow it to be counted for matches". It's worth discussing whether these properties should be merged, resulting in a simpler interface. I think there is a strong argument for merging them. In particular, I think it could be confusing to click on the diagram, and to have something happen which you can't connect with a cell in your signature, because the corresponding cell has been hidden.

**Comment by cdgls** [comment]
I agree those are two things you might want to do, but so far, I haven't needed to disable cells, I only want to be able to hide them.  This morning, on the eversion project, I'm spending a ton of time scrolling back and forth and back and forth across hundreds of cells that aren't relevant for what I'm doing.

I definitely think it would not be good if there were only one feature and it had the effect of hiding and disabling.  For instance, once you have finished say building all the 2-cells and you are working on building some 3-cells, you might want to hide all the 2-cells so you can focus on which 3-cells you are building, and just build using the click mechanism --- but you would be stuck if suddenly none of the 2-cells work at all.

**Comment by jamievicary** [comment]
OK, makes sense. So I'm thinking on the top row, to the left of the cell
name, a little tickbox to disable it, and a little '-' sign to minimize it.

**Comment by jamievicary** [comment]
These should NOT go next to the 'X', for the obvious reason.

**Comment by cdgls** [comment]
Yes, something like that sounds good.  Actually: I suggest putting these right where the X is now, and having the X be less accessible --- for instance, what if a right-click on a cell pulled up a pop-up and one of the options was "Delete Cell" --- I think that's more the amount of work that should be required for the relatively rare case of deleting a cell, and also has the advantage that it clears prime realestate for what I think will be the much much more often used minimize-unminimize cell.

**Comment by jamievicary** [comment]
OK, makes sense.

Realistically there are more bells and whistles we're going to want to add
to the cell area. I think there's going to have to be a 'settings' button
(perhaps a little cog icon), which when clicked expands the cell area, and
makes more options visible. 'Delete' can be hidden away there.

There are other weird things that one might want to put here that can't
currently be done any other way. For example, suppose you have two cells of
an identical type: you could ask Globular to merge them. It's an operation
that's perfectly mathematically meaningful, but cannot be performed using
other features.
